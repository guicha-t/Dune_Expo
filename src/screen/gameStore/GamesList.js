import React, { Component } from 'react';
import { Alert, Button, TextInput, View, Text,
  StyleSheet, AsyncStorage, ListView, TouchableOpacity,
  Picker, Item, FlatList, Image, Keyboard} from 'react-native';
  import { observer } from 'mobx-react';
  import { Avatar } from 'react-native-elements';

  import GridView from 'react-native-super-grid';

  import Header from './../../global/header/Header';
  import Store from './../../global/store/Store'

  @observer
  export default class GamesList extends Component {
    constructor(props){
      super(props);
      this.state = {
        GameList: [],
        Search: '',
	    Viewtype: '0',
      }
    }

    componentDidMount(){
        fetch('http://api.dune-table.com/v1/store/', {
                method: 'POST',
                Accept: 'application/json',
                headers: {
                  'Content-Type': 'application/json',
                  token: Store.Token,
                },
                body: JSON.stringify({
                  idType: 0,
                }),
              }).then((response) => response.json())
              .then((responseJson) => {
                this.setState({'GameList':responseJson.response})
              })
              .catch((error) => {
                console.error(error);
              });
    }

    _goToGameProfil = async (param) => {
      this.props.navigation.navigate('GameContainer', {
        id:param,
      });
    };

    _setCurrentGame = async (param) => {
    if (param.key === 'Enregistrées'){
    fetch('http://api.dune-table.com/v1/store/getAppsEcole', {
          method: 'GET',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
            token: Store.Token,
          },
        }).then((response) => response.json())
            .then((responseJson) => {
              this.setState({'GameList':responseJson.response})
            })
            .catch((error) => {
              console.error(error);
            });
      }

    else if (param.key === 'Dune Store'){
    fetch('http://api.dune-table.com/v1/store/', {
            method: 'POST',
            Accept: 'application/json',
            headers: {
              'Content-Type': 'application/json',
              token: Store.Token,
            },
            body: JSON.stringify({
              idType: 0,
            }),
          }).then((response) => response.json())
          .then((responseJson) => {
            this.setState({'GameList':responseJson.response})
          })
          .catch((error) => {
            console.error(error);
          });
        }
    }

    _searchRequest = async () => {
      Keyboard.dismiss()
              fetch('http://api.dune-table.com/v1/store/', {
                method: 'POST',
                Accept: 'application/json',
                headers: {
                  'Content-Type': 'application/json',
                  token: Store.Token,
                },
                body: JSON.stringify({
                  search: this.state.Search,
                  idType: 0,
                }),
              }).then((response) => response.json())
              .then((responseJson) => {
                this.setState({'GameList':responseJson.response})
              })
              .catch((error) => {
                console.error(error);
              });
    }

    render() {

      return (
        <View style={styles.mainContainer}>
          <Header navigation={this.props.navigation}/>
          <View style={styles.classContainer}>
            <View style={{flex: 1}}>
              <FlatList
                horizontal={true}
                showsHorizontalScrollIndicator={false}
                data={[{key: 'Enregistrées'}, {key: 'Dune Store'}]}
                showsVerticalScrollIndicator={false}
                renderItem={({item}) =>
                <View style={{width: 178}}>
                  <TouchableOpacity style={{ flex: 1 }} onPress={() => this._setCurrentGame(item)}>
                    <View style={styles.buttonClass}>
                        <Text style={styles.textClass}>{item.key}</Text>
                    </View>
                  </TouchableOpacity>
                </View>
              }
              />
          </View>
        </View>

        <View style={styles.searchContainer}>
          <TextInput
            value={this.state.Search}
            onChangeText={(Search) => this.setState({ Search })}
            placeholder={'Rechercher'}
            autoCapitalize = 'none'
            style={styles.input}
          />
          <Button
            title={'Go'}
            style={styles.ButtonSearch}
            color='#363453'
            onPress={() => this._searchRequest()}
          />
        </View>


        <View style={{flex: 0.8}}>
          <GridView
            itemDimension={100}
            spacing={1}
            items={this.state.GameList}
            style={styles.GridView}
            renderItem={item => (
              <View style={styles.itemContainer}>
                <TouchableOpacity style={{flex: 1}} onPress={() => this._goToGameProfil(item.id)}>
                  <View style={{flex: 1, marginLeft: 10, marginRight: 10}}>
                    <View style={{flex: 0.7, paddingTop: 10}}>
                      <Image
                        style={{flex: 1, borderRadius:10}}
                        source={{uri: 'http://176.31.252.134:9001/files/apps/' + item.picPath}}
                      />
                    </View>
                    <View style={{flex: 0.3, justifyContent: 'center', alignItems: 'center'}}>
                      <Text style={styles.itemName}>{item.nomApp}</Text>
                      <Text style={styles.itemName}>{item.nomCreator}</Text>
                    </View>
                  </View>
                </TouchableOpacity>
              </View>
            )}
            />
        </View>
      </View>
     );
  }
}

const styles = StyleSheet.create({


  container: {
   flex: 1,
   paddingTop: 22
  },
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },


  mainContainer: {
    flex:1,
    backgroundColor: '#fff',
  },
  classContainer: {
    flex: 0.1,
    flexDirection: 'row',
    paddingTop: 2,
  },
  searchContainer: {
    flex: 0.1,
    flexDirection: 'row',
    paddingTop: 2,
    alignItems: 'center',
    margin: 5,
  },
  input: {
    flex: 1,
    height: 44,
    padding: 10,
    marginRight: 5,
    borderWidth: 1,
    borderColor: 'black',
  },
  ButtonSearch: {
    padding: 10,
    borderWidth: 1,
    borderColor: 'black',
    marginBottom: 10,
  },
  allClassContainer: {
    flex:0.2,
  },
  gridView: {
    paddingTop: 25,
    flex: 1,
  },
  itemContainer: {
    flex: 1,
    height: 160,
    backgroundColor: '#FFF',
    borderRadius:10,
  },
  itemName: {
    fontSize: 12,
    color: '#434B77',
    fontWeight: '600',
  },
  buttonClassAll: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#FEE599',
    margin:2,
  },
  buttonClass: {
    width:178,
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    margin: 2,
    backgroundColor: '#FEE599',
    borderRadius:10,
  },
  textClass: {
    fontSize: 14,
    color: '#434B77',
    fontWeight: '600',
  },
});
