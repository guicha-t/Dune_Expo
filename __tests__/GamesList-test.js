import GamesList from './../src/screen/gameStore/GamesList';
import 'isomorphic-fetch';
import React from 'react';

import renderer from 'react-test-renderer'

jest.useFakeTimers();

test('render GamesList correctly', () => {

    const tree = renderer.create(<GamesList />).toJSON();
    expect(tree).toMatchSnapshot();
});