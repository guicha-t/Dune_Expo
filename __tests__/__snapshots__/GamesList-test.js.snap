// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render GamesList correctly 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "black",
          "elevation": 5,
          "flexDirection": "row",
          "height": 56,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
        },
        Object {
          "backgroundColor": "#fcebb5",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 0.3,
          "justifyContent": "center",
          "paddingLeft": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 24,
              "width": 24,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.4,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "flex": 1,
              "height": undefined,
              "width": undefined,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 0.3,
          "justifyContent": "center",
          "paddingRight": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 32,
              "width": 32,
            }
          }
        />
      </View>
    </View>
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderColor": "#ea4335",
              "borderRadius": 5,
              "borderWidth": 1,
              "marginHorizontal": 30,
              "maxWidth": 350,
              "shadowColor": "#000000",
              "shadowOpacity": 0.1,
              "shadowRadius": 10,
              "transform": Array [
                Object {
                  "scale": 0,
                },
              ],
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "justifyContent": "center",
                  "padding": 20,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333",
                      "fontSize": 18,
                      "fontWeight": "500",
                      "lineHeight": 25,
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                ATTENTION !
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#666",
                      "fontSize": 17,
                      "paddingTop": 10,
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                Vous devez accepter les conditions d'utilisation avant de pouvoir utiliser l'application.
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderRadius": 5,
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ea4335",
                    "borderRadius": 5,
                    "marginBottom": 20,
                    "marginHorizontal": 10,
                    "minWidth": 90,
                    "opacity": 1,
                    "paddingVertical": 8,
                  }
                }
                testID="buttonConfirm"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFF",
                        "fontSize": 15,
                        "fontWeight": "600",
                        "textAlign": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  COMPRIS
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "flex": 0.1,
        "flexDirection": "row",
        "paddingTop": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "justifyContent": "center",
          "width": 60,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#2A5E85",
                  "borderColor": "white",
                  "borderRadius": 30,
                  "borderWidth": 2,
                  "paddingLeft": 20,
                  "width": 60,
                },
                undefined,
                undefined,
              ]
            }
          >
            <Text />
            <Text
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fcebb5",
                "borderRadius": 10,
                "flex": 1,
                "justifyContent": "center",
                "margin": 2,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#2A5E85",
                  "fontSize": 14,
                  "fontWeight": "600",
                }
              }
            >
              Enregistrées
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fcebb5",
                "borderRadius": 10,
                "flex": 1,
                "justifyContent": "center",
                "margin": 2,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#2A5E85",
                  "fontSize": 14,
                  "fontWeight": "600",
                }
              }
            >
              Dune Store
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "flex": 0.15,
        "flexDirection": "row",
        "width": 750,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "overflow": "hidden",
            "paddingTop": 16,
          },
          Object {
            "backgroundColor": "#FFF",
            "color": "#FFF",
            "width": 680,
          },
          Object {
            "height": 64,
          },
        ]
      }
    >
      <Text />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#f0f0f0",
              "position": "absolute",
              "top": 8,
              "width": 1,
            },
            Object {
              "height": 48,
              "left": 48,
            },
          ]
        }
      />
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 48,
            "left": 56,
            "position": "absolute",
            "top": 24,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2A5E85",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Chercher un jeu
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        animationDuration={300}
        height={48}
        iconClass={[Function]}
        iconColor="#2A5E85"
        iconName="puzzle-piece"
        iconSize={20}
        iconWidth={40}
        inputPadding={16}
        label="Chercher un jeu"
        labelStyle={
          Object {
            "color": "#2A5E85",
          }
        }
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        passiveIconColor="#a3a3a3"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "padding": 7,
              "paddingLeft": 0,
            },
            Object {
              "color": "#2A5E85",
              "marginLeft": 56,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderRadius": 10,
                  "height": 60,
                  "justifyContent": "center",
                  "paddingLeft": 10,
                  "width": 60,
                },
                undefined,
                undefined,
              ]
            }
          >
            <Text />
            <Text
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "flex": 0.8,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2A5E85",
          "fontSize": 20,
        }
      }
    >
      Aucune application
    </Text>
  </View>
</View>
`;
